RockLib.Configuration.ObjectFactory/ConfigReloadingProxyFactory.cs

```diff
@@ -11,395 +11,409 @@
-            var baseGetObjectMethod = baseClass.GetTypeInfo().GetProperty("Object").GetMethod;
-            var baseTransferStateMethod = baseClass.GetTypeInfo().GetMethod("TransferState", BindingFlags.NonPublic | BindingFlags.Instance);
-
-            var eventFields = new Dictionary<EventInfo, FieldBuilder>();
-            var implementedMethods = new List<MethodInfo>();
-
-            var proxyTypeBuilder = CreateProxyTypeBuilder(interfaceType, baseClass);
-
-            AddConstructor(proxyTypeBuilder, baseConstructor);
-
-            // As a side-effect, AddProperty adds getter and setter methods to the implementedMethods list.
-            foreach (var property in interfaceType.GetAllProperties())
-                AddProperty(proxyTypeBuilder, property, baseGetObjectMethod, implementedMethods);
-
-            // As a side-effect, AddEvent adds:
-            // - add and remove event handler methods to the implementedMethods list.
-            // - the interface event + backing event handler field to the eventFields dictionary.
-            foreach (var evt in interfaceType.GetAllEvents())
-                AddEvent(proxyTypeBuilder, evt, baseGetObjectMethod, eventFields, implementedMethods);
-
-            // Only add methods that weren't already created in AddProperty and AddEvent.
-            foreach (var method in interfaceType.GetAllMethods().Where(method => !implementedMethods.Contains(method)))
-                AddMethod(proxyTypeBuilder, method, baseGetObjectMethod);
-
-            // The eventFields dictionary needs to be fully populated in order to correctly
-            // implement the TransferState method.
-            AddTransferStateOverrideMethod(proxyTypeBuilder, interfaceType, eventFields, baseTransferStateMethod);
-
-            return proxyTypeBuilder.CreateTypeInfo();
-        }
-
-        private static TypeBuilder CreateProxyTypeBuilder(Type interfaceType, Type baseType)
-        {
-            var assemblyName = $"<{interfaceType.Name}>a__RockLibDynamicAssembly";
-            var name = $"<{interfaceType.Name}>c__RockLibConfigReloadingProxyClass";
-            var assemblyBuilder = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName(assemblyName), AssemblyBuilderAccess.Run);
-            var moduleBuilder = assemblyBuilder.DefineDynamicModule("MainModule");
-            return moduleBuilder.DefineType(name, ProxyClassAttributes, baseType, new[] { interfaceType });
-        }
-
-        private static void AddConstructor(TypeBuilder proxyTypeBuilder, ConstructorInfo baseConstructor)
-        {
-            var constructorBuilder = proxyTypeBuilder.DefineConstructor(ConstructorAttributes, baseConstructor.CallingConvention,
-                new[] { typeof(IConfiguration), typeof(DefaultTypes), typeof(ValueConverters), typeof(Type), typeof(string), typeof(IResolver) });
-
-            var il = constructorBuilder.GetILGenerator();
+      // TODO: Can these really be null?
```

Comment:
Is this just a note for the next time we go through the libraries, or did you mean to do something with this?

Reply:
That comment was resolved, I just forgot to remove it (and now I have).